### PR TITLE
added a unit test for function injectionDisabled

### PR DESCRIPTION
**Pull request by visheshtha**
**Please provide a description of this PR:**
While going through `injector-list.go`,found out that there was no unit test case available for `injectionDisabled` function so  added a unit testcase for `injectionDisabled` function in `injector-list_test.go`.

### CODE REVIEW

**Comment by visheshtha** [comment]
Hi, @howardjohn,PTAL!

**Review by bleggett — COMMENTED** [review]
@visheshtha 

Just add 2 tests:

```
		{
			name: "Injection enabled by annotation",
			pod: &corev1.Pod{
				ObjectMeta: metav1.ObjectMeta{
					Annotations: map[string]string{
						annotation.SidecarInject.Name: "true",
					},
				},
			},
			expected: false,
		},
		{
			name: "Injection enabled by label",
			pod: &corev1.Pod{
				ObjectMeta: metav1.ObjectMeta{
					Annotations: map[string]string{
						annotation.SidecarInject.Name: "false",
					},
					Labels: map[string]string{
						label.SidecarInject.Name: "true",
					},
				},
			},
			expected: false,
		},
```

delete the others, they don't meaningfully improve branch coverage.

Code coverage without your tests:

```
go test -cover ./istioctl/pkg/injector/...  
ok      istio.io/istio/istioctl/pkg/injector    0.026s  coverage: 9.9% of statements

```
Code coverage with all the tests you added: 

```
go test -cover ./istioctl/pkg/injector/...
ok      istio.io/istio/istioctl/pkg/injector    0.026s  coverage: 12.6% of statements
```

Code coverage with just the two tests you added above: 

```
go test -cover ./istioctl/pkg/injector/...
ok      istio.io/istio/istioctl/pkg/injector    0.023s  coverage: 12.6% of statements
```

**Comment by visheshtha** [comment]
> @visheshtha
> 
> Just add 2 tests:
> 
> ```
> 		{
> 			name: "Injection enabled by annotation",
> 			pod: &corev1.Pod{
> 				ObjectMeta: metav1.ObjectMeta{
> 					Annotations: map[string]string{
> 						annotation.SidecarInject.Name: "true",
> 					},
> 				},
> 			},
> 			expected: false,
> 		},
> 		{
> 			name: "Injection enabled by label",
> 			pod: &corev1.Pod{
> 				ObjectMeta: metav1.ObjectMeta{
> 					Annotations: map[string]string{
> 						annotation.SidecarInject.Name: "false",
> 					},
> 					Labels: map[string]string{
> 						label.SidecarInject.Name: "true",
> 					},
> 				},
> 			},
> 			expected: false,
> 		},
> ```
> 
> delete the others, they don't meaningfully improve branch coverage.
> 
> Code coverage without your tests:
> 
> ```
> go test -cover ./istioctl/pkg/injector/...  
> ok      istio.io/istio/istioctl/pkg/injector    0.026s  coverage: 9.9% of statements
> ```
> 
> Code coverage with all the tests you added:
> 
> ```
> go test -cover ./istioctl/pkg/injector/...
> ok      istio.io/istio/istioctl/pkg/injector    0.026s  coverage: 12.6% of statements
> ```
> 
> Code coverage with just the two tests you added above:
> 
> ```
> go test -cover ./istioctl/pkg/injector/...
> ok      istio.io/istio/istioctl/pkg/injector    0.023s  coverage: 12.6% of statements
> ```

sure @bleggett,will update the suggested changes

**Comment by visheshtha** [comment]
Hi @bleggett,removed the testcases which are not relevant,PTAL!

File: istioctl/pkg/injector/injector-list_test.go [hunk 1]
```diff
@@ -105,3 +105,43 @@ func Test_getNamespaces(t *testing.T) {
...
+					},
+				},
+			},
+			expected: false,
+		},
+	}
+
+	for i, c := range cases {
+		t.Run(fmt.Sprintf("case %d %s", i, c.name), func(t *testing.T) {
+			assert.Equal(t, c.expected, injectionDisabled(c.pod))
```
Comment (hzxuzhonghu) [inline istioctl/pkg/injector/injector-list_test.go:144]: do you want to test disabled, why these two cases are both `enabled`

Comment (visheshtha) [reply istioctl/pkg/injector/injector-list_test.go:144]: > do you want to test disabled, why these two cases are both `enabled`

hi @hzxuzhonghu,I previously included all the testcases with enabled and disabled both which includes all the conditions,but they were providing the same code coverage as these two only so removed.
Do I need to cover all the testcases?

Comment (hzxuzhonghu) [reply istioctl/pkg/injector/injector-list_test.go:144]: We donot only care about coverage reported, we should look at the corner cases it coverred

Comment (visheshtha) [reply istioctl/pkg/injector/injector-list_test.go:144]: yes @hzxuzhonghu,agree with you,PTAL 7a5f0ff4d4a45e3e1622ad680c3444bcd91d5f3f,I have tried to cover most of the testcases,do I need to add more,if not,I will revert the changes.

**Comment by visheshtha** [comment]
Hi @hzxuzhonghu,included all the possible cases,PTAL!
or do we need anything more?

**Review by hzxuzhonghu — APPROVED** [review]

File: istioctl/pkg/injector/injector-list_test.go (hunk 1 above)
Comment (bleggett) [reply istioctl/pkg/injector/injector-list_test.go:231]: > We donot only care about coverage reported, we should look at the corner cases it coverred

Agree, this is why I wish `go test` wasn't so crude in terms of coverage options and supported `branch` rather than `statement` coverage - branch coverage _is_ a good proxy for "corner cases".

In this case, we are adding simple unit tests to a simple equality function with no subtle behavior that needs to be fenced, so statement coverage is "good enough".
